### ISSUE TITLE
[redis input] field type conflict for master_last_io_seconds_ago

### ISSUE THREAD

**Issue by juanpaulo** [influxdata/telegraf#1929]
the value [master_last_io_seconds_ago](http://redis.io/commands/INFO) may change to [`-1`](https://github.com/antirez/redis/blob/c7a4e694ad3689d934897b0d2c37144d7b2d0b97/src/server.c#L3047) on failover which would trigger the following error (since the value is explicitly specified as integer type initially, master_last_io_seconds_ago=0i -> master_last_io_seconds_ago=-1):

```
ERROR: {"error":"field type conflict: input field \"master_last_io_seconds_ago\" on measurement \"redis\" is type float64, already exists as type integer"}
```

This error will cause an output write error, in my case to InfluxDB, which would **affect writes of other gathered metrics as well**.
### Relevant telegraf.conf:

```
[[inputs.redis]]
  servers = ["tcp://localhost:6379"]
```
### System info:
- Telegraf v1.0.0
- InfluxDB v0.13.0
- CentOS 7.2
- Redis 3.2.1
### Steps to reproduce:
1. Start telegraf
2. Trigger redis failover
### Expected behavior:

Continue writing to output (from monitoring perspective)
### Actual behavior:

Error writing to output
### Additional info:

```
2016/10/24 02:18:20 Input [kernel] gathered metrics, (10s interval) in 270.302µs
2016/10/24 02:18:20 Input [mem] gathered metrics, (10s interval) in 909.017µs
2016/10/24 02:18:20 Input [swap] gathered metrics, (10s interval) in 2.939741ms
2016/10/24 02:18:20 Input [disk] gathered metrics, (10s interval) in 2.830161ms
2016/10/24 02:18:20 Input [cpu] gathered metrics, (10s interval) in 3.359479ms
2016/10/24 02:18:20 Input [net] gathered metrics, (10s interval) in 6.65063ms
2016/10/24 02:18:20 Input [processes] gathered metrics, (10s interval) in 30.553254ms
2016/10/24 02:18:20 Input [system] gathered metrics, (10s interval) in 33.632587ms
2016/10/24 02:18:20 Input [diskio] gathered metrics, (10s interval) in 33.324162ms
2016/10/24 02:18:20 Output [influxdb] buffer fullness: 67 / 10000 metrics. Total gathered metrics: 67. Total dropped metrics: 0.
2016/10/24 02:18:20 ERROR: {"error":"field type conflict: input field \"master_last_io_seconds_ago\" on measurement \"redis\" is type float64, already exists as type integer"}
2016/10/24 02:18:20 Error writing to output [influxdb]: Could not write to any InfluxDB server in cluster
2016/10/24 02:18:20 Input [redis] gathered metrics, (10s interval) in 356.22263ms
```
### Proposal:

Change field type of `master_last_io_seconds_ago` to float64.

**Comment by sparrc** [comment]
I think the correct solution is for that field to be sent as an integer. Integers can be negative afaik.

**Comment by sparrc** [comment]
> Trigger redis failover

can you explain how to do this?

**Comment by juanpaulo** [comment]
> I think the correct solution is for that field to be sent as an integer

Read a bit more about [InfluxDB's line protocol](https://docs.influxdata.com/influxdb/v1.0/write_protocols/line_protocol_reference/) and I agree on your solution!

Will look more into how to trigger redis failover. In our case, it just happened and the agent on that server started sending `master_last_io_seconds_ago=-1`. Should figure out how to make it `master_last_io_seconds_ago=-1i` which I believe would solve this issue.

**Comment by sparrc** [comment]
this was due to us trying to parse all redis metrics as unsigned integers, then falling back to floats, which meant that negative integers would get parsed as floats.

so this behavior has already been fixed in 1.1.0 with https://github.com/influxdata/telegraf/commit/41a5ee65713c644a2571267f5ee2a6c93926e3b0
